pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -65,6 +77,37 @@ func (s *BlockBuilderScheduler) stopping(_ error) error {
 }
 
 func (s *BlockBuilderScheduler) running(ctx context.Context) error {
+	// The startup process for block-builder-scheduler entails learning the state of the world:
+	//  1. obtain an initial set of offset info from Kafka
+	//  2. listen to worker updates for a while to learn what the previous scheduler knew
+	// When both of those are complete, we transition from observation mode to normal operation.
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		lag, err := s.fetchLag(ctx)
+		if err != nil {
+			panic(err)
+		}
+		s.committed = commitOffsetsFromLag(lag)
+	}()
+	go func() {
+		defer wg.Done()
+		time.Sleep(s.cfg.StartupObserveTime)
```

Comment:
I'm not sure how it will look in the later PRs, but maybe add a TODO here to bail from sleeping if the context is cancelled?